setup_cares.py

```diff
@@ -152,7 +153,23 @@ def build_extensions(self):
         if use_system_lib:
             self.compiler.add_library('cares')
         else:
+            self.compiler.define_macro('CARES_THREADS', 1)
             self.compiler.define_macro('CARES_STATICLIB', 1)
+            if hasattr(select, 'poll'):
+                self.compiler.define_macro('HAVE_POLL', 1)
+                self.compiler.define_macro('HAVE_POLL_H', 1)
+            if hasattr(select, 'kqueue'):
+                self.compiler.define_macro('HAVE_KQUEUE', 1)
+                self.compiler.define_macro('HAVE_SYS_TYPES_H', 1)
+                self.compiler.define_macro('HAVE_SYS_EVENT_H', 1)
+                self.compiler.define_macro('HAVE_SYS_TIME_H', 1)
+                self.compiler.define_macro('HAVE_FCNTL_H', 1)
+            if hasattr(select, 'epoll'):
+                self.compiler.define_macro('HAVE_EPOLL', 1)
+                self.compiler.define_macro('HAVE_SYS_EPOLL_H', 1)
+                self.compiler.define_macro('HAVE_FCNTL_H', 1)
+            if hasattr(select, 'select') and sys.platform != 'win32':
```

Comment:
Does Windows need no define?

Reply:
Correct, windows has its own selector always so when its on windows the internals of c-ares always uses that

Reply:
https://github.com/c-ares/c-ares/blob/6b76a55a91cfa5cdf56b7e743e45d8bae9a7e5a7/src/lib/ares_event.h#L178

Reply:
Maybe this is better to check if `os.pipe` exists

Reply:
adjusted per the comment in https://github.com/c-ares/c-ares/blob/6b76a55a91cfa5cdf56b7e743e45d8bae9a7e5a7/src/lib/ares_event.h#L182

Reply:
Left the ! win32 since c-ares wants a posix `pipe()` and not the windows implementation... plus its always going to use the win32 built-in event option anyways.